### DISCUSSION TITLE
Showing pieces in widgets - how?

### DISCUSSION THREAD

**Discussion by LeviticusMaximus** [apostrophecms/apostrophe#3851] [General]
In Apostrophe 2 I would create a piece and then I could use either a show/index page **or a widget** or both to display those pieces.

**There doesn't seem to be anything in the A3 docs** to tell me how to implement a widget to show pieces, nor if/why the feature has been removed.

I've made a _piece_ and a _piece-widget_, I've called them review and review-widget.

I added some review pieces. I then put the widget in my page. All seems well, but nothing shows up in the widget where I used to be able to select pieces individually, by tag, or show all.

The only things I can extend seem to be:
```
'@apostrophecms/piece-type'
'@apostrophecms/page-type'
'@apostrophecms/widget-type'
```

...and no pieces-widget-type for example.

Am I being blind, going mad, or missing something obvious?

**Comment by LeviticusMaximus** [comment]
I get it now, and it's a great way of doing it, although I reckon it would be advisable to make a section in the docs covering this functionality, or if it's there already then make it easily findable.

All I had to do was add a `_piece` field in the widget using the `relationship` type, for example in my case:

```
// review-widget/index.js

module.exports = {
  extend: '@apostrophecms/widget-type',
  options: {
    label: 'Review Widget'
  },
  fields: {
    add: {
      _review: {
        type: 'relationship',
        label: 'Review',
        withType: 'review',
        builders: {
          project: {
            title: 1,
            _url: 1,
            reviewer: 1,
            reviewText: 1,
            reviewType: 1
          }
        }
      }
    }
  }
};
```

**Comment by boutell** [comment]
cc @BoDonkey, it would make sense to make sure our pieces docs talk about how to make a widget that displays pieces by using a relationship field, since we no longer provide one by default. (We stopped because the alternatives to a simple relationship field were hardly ever used.)

**Comment by BoDonkey** [comment]
EDITED:
I can't seem to find this PR and I'm not satisfied with the documentation even if this PR landed. This will be revisited in a new ticket.
ORIGINAL:
Closed with PR#230. Thanks @LeviticusMaximus!
